File: 백준/Silver/7785. 회사에 있는 사람/회사에 있는 사람.py
n = int(input())
dic = {}
for _ in range(n):
    name, log = input().split()
    dic[name] = log

result = []
for k,v in dic.items():
    if v == 'enter':
        result.append(k)
print(*sorted(result, reverse=True),sep='\n')

File: 백준/Silver/7785. 회사에 있는 사람/회사에 있는 사람.swift
import Foundation

let n = Int(readLine()!)!
var dic = [String: Bool]()
for _ in 0..<n {
    let input = readLine()!.split(separator: " ").map { String($0) }
    if input[1] == "enter" {
        dic[input[0]] = true
    } else {
        dic.removeValue(forKey: input[0])
    }
}

var answer = ""
let keys = dic.keys.sorted { $0 > $1 }
for key in keys {
    answer += "\(key)\n"
}
print(answer)